pennylane/utils.py

```diff
@@ -323,6 +326,22 @@ def circuit2():
             + ",".join(string_reps)
         )
 
+    if qml.tape_mode_active():
+        for op in operation_list:
```

Comment:
So usually `operation_list` is the queue recorded by the `OperationRecorder` right? But if this is an invisible recorder, how come you have to remove it from the `QueuingContext`?

Reply:
> So usually operation_list is the queue recorded by the OperationRecorder right?

The `operation_list` will in most cases be recorded by a `QuantumTape`, which will keep a record of the original un-inverted operations. For example,

```python
@qml.qnode(dev)
def circuit(x):
    qml.RX(x, wires=0)
    qml.inv([qml.RY(y, wires=1), qml.RZ(wires=0)])
    return qml.expval(qml.PauliZ(0))
```

The QNode is inserting an implicit tape under the hood to record the users operations. This is recording the RX, RY, and RZ gates. We then remove RY and RZ, invert, and re-queue.

Not ideal, but 'good enough' for now. I think this function is barely used, it should be re-born as a tape transform.

Reply:
I tested the inversion in a kernel pipeline and it seems to work great (at least the models train).